### ISSUE TITLE
API v2: List all entities

### ISSUE THREAD

**Issue by fgalan** [telefonicaid/fiware-orion#947]
Implement list all entities operation in v2, i.e. 

```
GET /v2/entities?params
```

See detailed description in the NGSIv2 APIary reference at http://telefonicaid.github.io/fiware-orion/api/v2/.

**Comment by kzangeli** [comment]
Just to add one little detail.

We have said that if **type** is empty for an attribute (+ no metadata), then the attribute is rendered as:

    "NAME": "VALUE"

Now, if the attribute if of type "number" (+ no metadata) then this special type is considered "empty" as well, and rendered as follows:

    "NAME": VALUE

The same is valid for metadata.

**Comment by fgalan** [comment]
Raised in https://github.com/telefonicaid/fiware-orion/pull/958/files#r31917180, right processing of Accept in /v2/entities operations is not part of PR #958 and would be addressed in a next PR (maybe developing an overall solution for all /v2/ operations, which are purely based on JSON).

**Comment by fgalan** [comment]
Support to location using `geo:point` is not part of PR #958 and would be addressed in a next PR.

**Comment by fgalan** [comment]
Finally, it is not part of PR #958 solving the ambiguenty problems for compound attributes e.g.:

What I mean is that for example the following rendering for attribute A1:

```
"A1": {
  "value": "M1", 
  "type": "a4", 
  "m2": {
    "type": "tm2", 
    "value": "M2"
   }, 
   "m3": 3,
}
```
Could be interpreted in two different ways:

* As an attribute without metadata and type and which value is the following JSON object:

```
{
  "value": "M1", 
  "type": "a4", 
  "m2": {
    "type": "tm2", 
    "value": "M2"
   }, 
   "m3": 3,
}
```

* As an attribute of type "a4" and value "M1" with 2 metadata (one named "m2" with type "tm2" and value "M2" and the other named "m3" of type "number" and value 3.

**Comment by fgalan** [comment]
After latest discussion, the `count` option (within the `options` parameter) will be used instead of `count=true`. The code needs to be fixed (PR #958 was based on `count=true`).

**Comment by fgalan** [comment]
The basic operation has been implemented in PR #958 and all "pending aspects" originally included here as comments have been moved to independent issues, to provide a cleaner task separation and enable better scheduling. Thus, this issue can be closed.
